src/managers/discord_client.py

```diff
@@ -0,0 +1,16 @@
+from discord.ext import commands
+
+
+class _DiscordClientManager(commands.Bot):
+    async def on_ready(self):
+        print(f"Logged on as {self.user}!")
+
+    async def on_message(self, message):
+        # Ensures commands work properly, wont process any commands without it.
+        await self.process_commands(message)
+
+        print(f"Message from {message.author}: {message.content}")
```

Comment:
I am still debating the purpose of overriding the class, I dont really know the benefits of doing so. Is this is a smart thing to do or should I leave it untouched?

Reply:
Its not about the `await` but about the `self.process_commands()`. Leaving it out will break any command, as they'd simply be unprocessed.

Sadly `on_message` is in charge of taking care of the commands too, however the [docs](https://discordpy.readthedocs.io/en/stable/api.html?highlight=on_message#discord.on_message) do *not* state this. 

Terrible design if you ask me. Anything prefixed with the `command_prefix` (bot prefix) should have its own event imo.

Reply:
Its not about the `await` but about the `self.process_commands()`. Leaving it out will break any command, as they'd simply be unprocessed.

Sadly `on_message` is in charge of taking care of the commands too, however the [docs](https://discordpy.readthedocs.io/en/stable/api.html?highlight=on_message#discord.on_message) do *not* state this. 

Terrible design if you ask me. Anything prefixed with the `command_prefix` (bot prefix) should have its own event imo.